### ISSUE TITLE
Discussion about tags to put in Java config

### ISSUE THREAD

**Issue by mirkoperillo** [exercism/java#1890]
Ref to #1866 we have to add some tags in the `config.json` of Java track.
In the reference issue there is a link to the docs repo containing all the available tags.

What are the most appropriated for the Java track ?

**Comment by ericbalawejder** [comment]
https://github.com/exercism/docs/blob/main/anatomy/tracks/config-json.md#tags
```
{
"tags": [
    "paradigm/declarative",
    "paradigm/functional",
    "paradigm/imperative",
    "paradigm/object_oriented",
    "typing/static",
    "typing/strong",
    "execution_mode/compiled",
    "execution_mode/interpreted",
    "platform/windows",
    "platform/mac",
    "platform/linux",
    "platform/andriod",
    "runtime/jvm",
    "used_for/backends",
    "used_for/cross_platform_development",
    "used_for/mobile",
    "used_for/artificial_intelligence",
    "used_for/games"
  ]
}
```
Not sure if `typing/strong` is a gray area for Java. `execution_mode` seems like it's both? Static type check that's compiled to byte code and then interpreted?

**Comment by mirkoperillo** [comment]
citing the ref page

> execution_mode/interpreted: code is interpreted directly
So it's not the case of Java

I'm not sure we can use `paradigm/declarative` for Java.

I'm agree with all the others tags

**Comment by ericbalawejder** [comment]
Ok, drop `execution_mode/interpreted`. I usually see `declarative` associated with the functional features of Java 1.8+. I feel like Java isn't weakly typed but it isn't strongly typed either. Maybe just drop the `typing/strong`?

**Comment by mirkoperillo** [comment]
well we can drop `typing/strong` for the moment waiting for a wider discussion

I think that a functional paradigm bring to a more declarative approach, but I think it's a strong concept define Java as declarative language. ( SQL is an example of declarative language)

**Comment by ericbalawejder** [comment]
https://github.com/exercism/docs/blob/main/anatomy/tracks/config-json.md#tags
```
{
"tags": [
    "paradigm/declarative",
    "paradigm/functional",
    "paradigm/imperative",
    "paradigm/object_oriented",
    "typing/static",
    "execution_mode/compiled",
    "platform/windows",
    "platform/mac",
    "platform/linux",
    "platform/andriod",
    "runtime/jvm",
    "used_for/backends",
    "used_for/cross_platform_development",
    "used_for/mobile",
    "used_for/artificial_intelligence",
    "used_for/games"
  ]
}
```
Just updating this for discussion. Still undecided on `paradigm/declarative`.

**Comment by mirkoperillo** [comment]
@msomji do you have some suggestions about ?

**Comment by mirkoperillo** [comment]
Taking inspiration from C# track, they have `paradigm/functional` but not the `paradigm/declarative` one

**Comment by ericbalawejder** [comment]
```
{
"tags": [
    "paradigm/functional",
    "paradigm/imperative",
    "paradigm/object_oriented",
    "typing/static",
    "execution_mode/compiled",
    "platform/windows",
    "platform/mac",
    "platform/linux",
    "platform/andriod",
    "runtime/jvm",
    "used_for/backends",
    "used_for/cross_platform_development",
    "used_for/mobile",
    "used_for/artificial_intelligence",
    "used_for/games"
  ]
}
```
This appears to be last last issue to resolve the linter check failures. Does @msomji have any final input for this discussion?

**Comment by mirkoperillo** [comment]
@ericbalawejder  we have to unblock this issue.
I have a proposal: what you think to provide a PR about this ?
In this moment it is important to have a first version of tags to avoid the linter error and to go on with v3 prerequisites

**Comment by ericbalawejder** [comment]
I'm on board with trying to complete it for the linter as well as the v3 track. I was just waiting to see if there was any additional input. I can't make a PR with the code from my last comment if that works?

**Comment by mirkoperillo** [comment]
> I can't make a PR with the code from my last comment if that works?

Absolutely yes

I think we can define these tags for now and then if there will be improvements, we will provide other PRs.

**Comment by ericbalawejder** [comment]
Ok, "can't" = "can". Sorry if that caused confusion.

**Comment by mirkoperillo** [comment]
fixed by #1906
